lib/charms/traefik_k8s/v0/ingress_per_unit.py

```diff
@@ -419,13 +421,70 @@ def is_failed(self, relation: Relation = None) -> bool:
 
         return False
 
-    def get_request(self, relation: Relation):
-        """Get the IngressRequest for the given Relation."""
-        provider_app_data, requirer_unit_data = self._fetch_relation_data(relation)
-        return IngressRequest(self, relation, provider_app_data, requirer_unit_data)
+    def is_unit_ready(self, relation: Relation, unit: Unit) -> bool:
+        """Report whether the given unit has shared data in its unit data bag."""
+        # sanity check: this should not occur in production, but it may happen
+        # during testing: cfr https://github.com/canonical/traefik-k8s-operator/issues/39
+        assert (
+            unit in relation.units
+        ), "attempting to get ready state for unit that does not belong to relation"
+        if relation.data.get(unit, {}).get("data"):
+            # TODO consider doing schema-based validation here
+            return True
+        return False
```

Comment:
```suggestion
        # TODO consider doing schema-based validation here
        return relation.data.get(unit, {}).get("data")
```

Reply:
I purposefully left it like so because I expect more stuff to go in the is_ready method. 
Also I really like a method called `is_ready` to return `bool` :)

Reply:
wrapping it in `bool()` would solve that, but I get your point.